src/js/messaging/reducers/folders.js

```diff
@@ -62,6 +65,8 @@ export default function folders(state = initialState, action) {
     case SET_CURRENT_FOLDER:
       // The + forces +action.folderId to be a number
       return set('data.currentItem.persistFolder', +action.folderId, state);
+    case SET_FOLDER_NAME:
+      return set('data.newFolderName', action.folderName, state);
```

Comment:
This should probably live in the modal reducer under `createNewFolder`. It is related with folders, but this state is more closely related with the `createNewFolder` modal itself.